### PR TITLE
Feature/side bar mobile

### PR DESCRIPTION
**Pull request by phongduong-dp**
# Description

Create sidebar component used on mobile device.

This component will render <SideBarMobileItem/> as children component.

Depend on whether subItems are provided, <SideBarMobileItem/> will be rendered recursively.

Assume sideBarItems are provided as below:

```typescript
const items: SideBarMobileItemProps[] = [
  {
    title: "Trang chủ",
    url: "/",
  },
  {
    title: "Giới thiệu",
    url: "/about",
  },
  {
    title: "Danh sách sản phẩm",
    url: "/products",
    subItems: [
      {
        title: "Đầu đốt",
        url: "/#",
        subItems: [
          {
            title: "Đầu đốt theo hãng",
            url: "/#",
            subItems: [
              {
                title: "Madas",
                url: "/#",
              },
              {
                title: "Blowtherm",
                url: "/#",
              },
            ],
          },
        ],
      },
      {
        title: "Đầu đốt",
        url: "/#",
      },
    ],
  },
  {
    title: "Liên hệ",
    url: "/contact",
  },
];
```
Component will be used as follow:
```typescript
// handleONSideBarItemClick will redirect to another url using url properties
  <SideBarMobile
    closeButtonIcon={<CloseIcon />}
    sideBarItems={items}
    onCloseBtnClick={handleClosedBtnClick}
    onSideBarItemClick={handleOnSideBarItemClick}
    dropDownIcon={<DropDownIcon />}
  />
```

# Checklist

- [ ] This is a breaking changes?

# Video

https://github.com/its-rav/namvietstore-fe/assets/92157056/def769d6-1109-488d-a5e4-3bbada139f5b

### CODE REVIEW

**Comment by its-rav** [comment]
should this have full width and height?

**Comment by phongduong-dp** [comment]
> should this have full width and height?

Updated, I made this side bar fixed and full height, full width. 

https://github.com/its-rav/namvietstore-fe/assets/92157056/e3ac8f68-c077-4ef8-97e2-6ad1c84e515b

File: packages/core-components/src/components/SideBarMobile.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+};
+
+const SideBarMobile: React.FC<SideBarMobileProps> = ({
+  onCloseBtnClick,
+  onSideBarItemClick,
+  closeButtonIcon,
+  sideBarItems,
+  dropDownIcon,
+}) => {
+  return (
```
Comment (its-rav) [inline packages/core-components/src/components/SideBarMobile.tsx:80]: just wonder, why do we have border-radius for this, this looks weird to me

Comment (phongduong-dp) [reply packages/core-components/src/components/SideBarMobile.tsx:80]: I see it like this. Does this mean the border-radius ?1

Comment (its-rav) [reply packages/core-components/src/components/SideBarMobile.tsx:80]: nope. The design is using iphone layout I think, It should not be rounded

**Review by its-rav — APPROVED** [review]

File: packages/core-components/src/components/SideBarMobile.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  dropDownIcon,
+  subItems,
+}) => {
+  const [itemOpen, setItemOpen] = React.useState(false);
+
+  const openSubItems = () => {
+    setItemOpen(!itemOpen);
+  };
+
+  const renderSubItems = () => {
```
Comment (its-rav) [inline packages/core-components/src/components/SideBarMobile.tsx:31]: it's better to use useMemo to reduce unnecessary re-render @phongduong-dp FYI

**Pull request merged**